format.go

```diff
@@ -274,11 +289,11 @@ func (r *anyResolver) Resolve(typeUrl string) (proto.Message, error) {
 	if !ok {
 		return nil, fmt.Errorf("unknown message: %s", typeUrl)
 	}
-	// populate any extensions for this message, too
-	if exts, err := r.source.AllExtensionsForType(mname); err != nil {
-		return nil, err
-	} else if err := r.er.AddExtension(exts...); err != nil {
-		return nil, err
+	// populate any extensions for this message, too (if there are any)
+	if exts, err := r.source.AllExtensionsForType(mname); err == nil {
+		if err := r.er.AddExtension(exts...); err != nil {
+			return nil, err
+		}
```

Comment:
What's going on here?

Reply:
I believe this fixes https://github.com/fullstorydev/grpcurl/issues/193 but @erikjoh can confirm

Reply:
To give some extra context, we are trying to adopt the richer error model of grpc but grpcurl currently outputs the errors like this:

```
ERROR:
  Code: InvalidArgument
  Message: Request contains invalid fields
  Details:
  1)	{"@error":"google.rpc.BadRequest is not recognized; see @value for raw binary message data","@type":"type.googleapis.com/google.rpc.BadRequest","@value":"CiQKCnN0cmluZ19vbmUSFk5vIGVtcHR5IHZhbHVlIGFsbG93ZWQKJAoKc3RyaW5nX3R3bxIWTm8gZW1wdHkgdmFsdWUgYWxsb3dlZA=="}
```

The problem with the current code here is that when using service reflection to resolve `google.rpc.BadRequest`, it doesn't work since `google.rpc.BadRequest` doesn't have any protobuf extension.
You can see here that `r.source.AllExtensionsForType(mname)` returns an `err` `github.com/jhump/protoreflect/grpcreflect.elementNotFoundError` which then aborts the formatting.

![image](https://user-images.githubusercontent.com/16289015/227005246-3517e76c-db34-41a0-bfe1-72fa3f58fdef.png)

With the change in this PR, the `elementNotFoundError` will be ignored and the formatting of the error will successfully be processed and printed:

```
ERROR:
  Code: InvalidArgument
  Message: Request contains invalid fields
  Details:
  1)	{
    	  "@type": "type.googleapis.com/google.rpc.BadRequest",
    	  "field_violations": [
    	    {
    	      "field": "string_one",
    	      "description": "No empty value allowed"
    	    },
    	    {
    	      "field": "string_two",
    	      "description": "No empty value allowed"
    	    }
    	  ]
    	}
```

This new code should add extensions (i.e. `r.er.AddExtension(exts...)`) only if extensions could be found with `r.source.AllExtensionsForType(mname)`.

Reply:
Gotcha, that makes sense.  Why not also ignore the error from `r.er.AddExtension(exts...)`?